### PR TITLE
fix #9189 chore(project): use latest ruff rules

### LINKED ISSUE #9189: Use latest ruff rules
We recently switched to using [ruff](https://github.com/astral-sh/ruff) for Python linting, and it has since added many new helpful rules.  Let's update to the latest version of ruff and apply as many rules as we can.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-3716)

### PR DESCRIPTION
**Pull request by jaredlockhart**
Because

* The latest version of ruff has many more linting and auto formatting rules

This commit

* Updates to the latest ruff and applies many of its new linting and auto formatting rules

### CODE REVIEW

File: experimenter/experimenter/experiments/api/v5/serializers.py [hunk 1]
```diff
@@ -407,7 +407,6 @@ def to_representation(self, data):
 
         This enforces that the serialized data is ordered by the feature config ID.
         """
-        super().to_representation
```
Comment (mikewilli) [inline experimenter/experimenter/experiments/api/v5/serializers.py:410]: Is this not important?

**Review by mikewilli — APPROVED** [review]
Neat! Separate PRs for Cirrus and Schemas directories?

File: experimenter/experimenter/experiments/api/v5/serializers.py (hunk 1 above)
Comment (mikewilli) [reply experimenter/experimenter/experiments/api/v5/serializers.py:410]: Ah yea fair enough, good job ruff.

**Comment by jaredlockhart** [comment]
> 

Oh yeah I got started but there's a bunch of other infrastructural work that has to happen first so I'll just do it separately.

**Pull request merged**